### PR TITLE
MOB-3106: XCFramework Instructions

### PR DESCRIPTION
**Pull request by dvernon-splunk**
Adds a markdown file that contains instructions on how to create an XCFramework from the project.

### CODE REVIEW

**Review by seemk — APPROVED** [review]
Perhaps we should link to this file from the README as well?

I wonder once we have a new tag/release available, could we mention that we do build and distribute XCFrameworks? At the moment they are built during the CI and placed under artifacts: https://github.com/signalfx/splunk-otel-ios/actions/runs/5084261832

The artifacts will appear under a specific release, although currently it seems only tags are used.

File: frameworkInstructions.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# How to Build an XCFramework
```
Comment (theletterf) [inline frameworkInstructions.md:1]:
```suggestion
# How to build an XCFramework
```

File: frameworkInstructions.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+If you would like to import Splunk Otel as a framework into your project, follow these steps:
```
Comment (theletterf) [inline frameworkInstructions.md:3]:
```suggestion
If you want to import Splunk Otel as a framework into your project, follow these steps:
```

File: frameworkInstructions.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+## 1. Ensure the correct build settings.
```
Comment (theletterf) [inline frameworkInstructions.md:5]:
```suggestion
## 1. Check build settings
```

File: frameworkInstructions.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+Clone the splunk-otel-ios repo onto your local environment and open the SplunkRumWorkspace.xcworkspace file in XCode. Navigate to the Build Settings tab on the SplunkOtel target and ensure the following two settings are set:
+* Skip Install: No
+* Build Libraries for Distribution: Yes
```
Comment (theletterf) [inline frameworkInstructions.md:9]:
```suggestion
Clone the `splunk-otel-ios` repo and open the `SplunkRumWorkspace.xcworkspace` file in Xcode. Navigate to the **Build Settings** tab on the `SplunkOtel` target and make sure the following settings are present:

* Skip Install: No
* Build Libraries for Distribution: Yes
```

File: frameworkInstructions.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+## 2. Make Archives
```
Comment (theletterf) [inline frameworkInstructions.md:11]:
```suggestion
## 2. Make archives
```

File: frameworkInstructions.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+Open the terminal and navigate to the SplunkRumWorkspace/SplunkRum directory. You should be in the same directory as the SplunkRum.xcodeproj file. Paste the following code into the terminal.
+
+    xcodebuild archive -project SplunkRum.xcodeproj -scheme SplunkOtel -destination "generic/platform=iOS" -archivePath "archives/SplunkRum-iOS"
+
+This should create a new archives directory with the file `SplunkRum-iOS.xcarchive` inside it.
+
+Now repeat the process for the simulator platform:
+
+    xcodebuild archive -project SplunkRum.xcodeproj -scheme SplunkOtel -destination "generic/platform=iOS Simulator" -archivePath "archives/SplunkRum-iOS_Simulator"
```
Comment (theletterf) [inline frameworkInstructions.md:21]:
````suggestion
Open the terminal and navigate to the directory where the `SplunkRum.xcodeproj` file is located, for example `SplunkRumWorkspace/SplunkRum`. 

Run the following command to create a new archives directory with the file `SplunkRum-iOS.xcarchive` inside:

```bash
    xcodebuild archive -project SplunkRum.xcodeproj -scheme SplunkOtel -destination "generic/platform=iOS" -archivePath "archives/SplunkRum-iOS"
```

Repeat the process for the simulator platform:

```bash
    xcodebuild archive -project SplunkRum.xcodeproj -scheme SplunkOtel -destination "generic/platform=iOS Simulator" -archivePath "archives/SplunkRum-iOS_Simulator"
```
````

File: frameworkInstructions.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    
+## 3. Make XCFramework
+
+Paste the following code into the terminal:
```
Comment (theletterf) [inline frameworkInstructions.md:25]:
```suggestion
Run the following command to create a new archives directory with the file `SplunkRum-iOS.xcarchive`:
```

File: frameworkInstructions.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+    xcodebuild -create-xcframework -archive archives/SplunkRum-iOS.xcarchive -framework SplunkOtel.framework -archive archives/SplunkRum-iOS_Simulator.xcarchive -framework SplunkOtel.framework -output xcframeworks/SplunkOtel.xcframework
+
+This will create a new xcframeworks directory with the SplunkOtel.xcframework file inside of it.
```
Comment (theletterf) [inline frameworkInstructions.md:29]:
````suggestion
```bash
    xcodebuild -create-xcframework -archive archives/SplunkRum-iOS.xcarchive -framework SplunkOtel.framework -archive archives/SplunkRum-iOS_Simulator.xcarchive -framework SplunkOtel.framework -output xcframeworks/SplunkOtel.xcframework
```
````

File: frameworkInstructions.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+## 4. Importing the XCFramework into your project
+
+Open your project in XCode and drag and drop the SplunkOtel.xcframework into the project navigator. This should automatically import the framework.
```
Comment (theletterf) [inline frameworkInstructions.md:33]:
```suggestion
## 4. Import the XCFramework into your project

Open your project in Xcode and drag the `SplunkOtel.xcframework` file into the project navigator. This automatically imports the framework. 
```

**Review by theletterf — CHANGES_REQUESTED** [review]
Some docs edits required.

I think we could move this into a `development` folder. WDYT?

**Comment by theletterf** [comment]
@seemk These instructions are for instrumentation developers or for users? If they're for users, they should be in the user docs. Would a user build an XCFramework? Why?

**Comment by seemk** [comment]
> @seemk These instructions are for instrumentation developers or for users? If they're for users, they should be in the user docs. Would a user build an XCFramework? Why?

For advanced users, we've had requests for how to build the distro as an xcframework, but these are pretty rare cases

**Comment by theletterf** [comment]
@seemk If this is for customers, even if advanced, it should go into the official docs. I suggest we close this PR. I'll open a docs one in https://github.com/splunk/public-o11y-docs and link to this.

**Comment by theletterf** [comment]
Closing after checking with @dvernon-splunk — will move to the docs repo.

**Comment by jnino-splunk** [comment]
@theletterf where could I find where this lives now? Is it user accessible?

**Comment by theletterf** [comment]
@jnino-splunk It's in the docs PR we're building for GA: https://github.com/splunk/private-o11y-docs/pull/882/files

**Comment by jnino-splunk** [comment]
@theletterf  ah okay so not https://github.com/splunk/public-o11y-docs. I will have to wait for access to the private docs repo.

**Pull request closed without merging**